packages/app/src/app/overmind/effects/netlify.ts

```diff
@@ -0,0 +1,112 @@
+import axios from 'axios';
+import { Sandbox, NetlifySite } from '@codesandbox/common/lib/types';
+import getTemplate from '@codesandbox/common/lib/templates';
+import getNetlifyConfig from 'app/utils/getNetlifyConfig';
+
+const NetlifyBaseURL = 'https://netlify.deploy.codesandbox.io/site';
+
+type Options = {
+  getUserId(): string;
+};
+
+export default (() => {
+  let _options: Options;
+
+  return {
+    initialize(options: Options) {
+      _options = options;
+    },
+    async waitForDeploy(id: string, onLogUrlUpdate: (logUrl: string) => void) {
+      const url = `${NetlifyBaseURL}/${id}/status`;
+
+      const { data } = await axios.get(url);
+
+      if (data.status.status === 'IN_PROGRESS') {
+        await pollUntilDone(axios, url, 10000, 60 * 2000, onLogUrlUpdate);
+      }
+    },
+    async getDeployments(sandboxId: string): Promise<NetlifySite> {
+      const response = await axios.request({
+        url: `${NetlifyBaseURL}/${sandboxId}`,
+      });
+
+      return response.data;
+    },
+    async deploy(file: string, sandbox: Sandbox): Promise<string> {
+      const userId = _options.getUserId;
+      const template = getTemplate(sandbox.template);
+      const buildCommand = (name: string) => {
+        if (name === 'styleguidist') return 'styleguide:build';
+        if (name === 'nuxt') return 'generate';
```

Comment:
@christianalfoni Put curly braces around this one to make it less error-prone

```suggestion
        if (name === 'nuxt') {
          return 'generate';
        }
```